### PR TITLE
Tests: Account for array-like objects in grep()

### PR DESCRIPTION
**Pull request by AurelioDeRosa**
Reference: https://github.com/jquery/api.jquery.com/pull/801

### CODE REVIEW

**Comment by arthurvr** [comment]
Can anyone take a look at this PR? It seems pretty straightforward :)

**Comment by mzgol** [comment]
LGTM.

**Comment by markelog** [comment]
Would be cool to move it to different test

**Comment by markelog** [comment]
Friendly ping

**Comment by AurelioDeRosa** [comment]
Hi @markelog.

I'll do it tonight.

**Comment by AurelioDeRosa** [comment]
PR updated.

**Comment by mzgol** [comment]
@AurelioDeRosa The CLA check has failed. Perhaps you have incorrect data in one of the commits?

**Comment by AurelioDeRosa** [comment]
Fixed. Sorry about that.

File: test/unit/core.js [hunk 1]
```diff
@@ -956,17 +956,43 @@ QUnit.test( "jQuery.grep()", function( assert ) {
 
 	assert.deepEqual( jQuery.grep( [], searchCriterion ), [], "Empty array" );
 	assert.deepEqual( jQuery.grep( new Array( 4 ), searchCriterion ), [], "Sparse array" );
+	assert.deepEqual( jQuery.grep( { length: 0 }, searchCriterion ), [], "Empty array-like" );
 
 	assert.deepEqual( jQuery.grep( [ 1, 2, 3, 4, 5, 6 ], searchCriterion ), [ 2, 4, 6 ], "Satisfying elements present" );
 	assert.deepEqual( jQuery.grep( [ 1, 3, 5, 7 ], searchCriterion ), [], "Satisfying elements absent" );
+	assert.deepEqual( jQuery.grep( { 0: 1, 1: 2, 2: 3, 3: 4, 4: 5, 5: 6, length: 6 }, searchCriterion ), [ 2, 4, 6 ], "Satisfying elements present and array-like object used" );
```
Comment (mzgol) [inline test/unit/core.js]: These tests are now duplicated; they shouldn't be here at all.

File: test/unit/core.js [hunk 2]
```diff
@@ -967,6 +967,25 @@ QUnit.test( "jQuery.grep()", function( assert ) {
...
+QUnit.test( "jQuery.grep(Array-like)", function( assert ) {
+	assert.expect( 7 );
+
+	var searchCriterion = function( value ) {
+		return value % 2 === 0;
+	};
+
+	assert.deepEqual( jQuery.grep( { length: 0 }, searchCriterion ), [], "Empty array-like" );
+
+	assert.deepEqual( jQuery.grep( { 0: 1, 1: 2, 2: 3, 3: 4, 4: 5, 5: 6, length: 6 }, searchCriterion ), [ 2, 4, 6 ], "Satisfying elements present and array-like object used" );
```
Comment (markelog) [inline test/unit/core.js]: Could make that less then 120 symbols on the line?

Comment (markelog) [inline test/unit/core.js]: Sorry, for being so nitpicky btw

Comment (AurelioDeRosa) [inline test/unit/core.js]: @markelog Done. Please, let me know if that's OK. Also, you don't have to apologize. It's totally fine to do more work on the PR to improve its quality standard.

**Comment by markelog** [comment]
Landed! Thank you

**Pull request closed without merging**